# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1467758738 25200
#      Tue Jul 05 15:45:38 2016 -0700
# Node ID 21c12b837089d16bb52b5e5942bc85faa15c03af
# Parent  26c0e65e6b7751ea4a45a4492df7eb2b68ac0bc7
build: don't delete distributed docs on 'make clean' or 'make distclean' (bug #45646)

* Makefile.am (clean-aminfo): Disable built-in Automake target that deletes
Texinfo documentation on 'make clean'.
* doc/module.mk (doc-maintainer-clean): Delete all built Texinfo documentation
on 'make maintainer-clean'.
* doc/interpreter/module.mk (doc_DISTCLEANFILES): Remove $(OCTAVE_HTML_STAMP)
from the list.

diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -424,16 +424,22 @@ install-data-local: installdirs-local
 
 clean-local: doc-clean
 
 distclean-local:
 	if [ "x${srcdir}" != "x." ]; then rm -f HG-ID; fi
 
 maintainer-clean-local: doc-maintainer-clean
 
+## The 'clean-aminfo' target is defined by Automake >= 1.11.  We want to
+## distribute all Texinfo docs with the source distribution and not delete
+## them on 'clean', so we override this target to do nothing by default.
+
+clean-aminfo:
+
 define move_if_change_rule
   if [ -s $(1) ]; then \
     ${SHELL} ${top_srcdir}/build-aux/move-if-change $(1) $(2); \
   else \
     echo "$(1) is empty!" 1>&2; \
     rm -f $(1); \
     exit 1; \
   fi
diff --git a/doc/interpreter/module.mk b/doc/interpreter/module.mk
--- a/doc/interpreter/module.mk
+++ b/doc/interpreter/module.mk
@@ -345,19 +345,16 @@ doc_EXTRA_DIST += \
   doc/interpreter/mk_doc_cache.m \
   doc/interpreter/mkcontrib.awk \
   doc/interpreter/munge-texi.pl \
   $(DOC_IMAGES) \
   $(DOC_IMAGES_SRC) \
   $(LOGOS) \
   $(TXI_SRC)
 
-doc_DISTCLEANFILES += \
-  $(OCTAVE_HTML_STAMP)
-
 doc_MAINTAINERCLEANFILES += \
   AUTHORS \
   $(BUILT_DOC_IMAGES) \
   $(BUILT_OCTAVE_TEXI_SRC) \
   doc/interpreter/doc-cache
 
 ## The TeX software suite is used to create both PDF and PS output formats.
 ## In order to avoid race conditions between simultaneous TeX commands, the
diff --git a/doc/module.mk b/doc/module.mk
--- a/doc/module.mk
+++ b/doc/module.mk
@@ -64,8 +64,9 @@ doc-clean: doc-interpreter-clean doc-lib
 	rm -f $(doc_CLEANFILES)
 	rm -rf doc/.texmf-var
 
 doc-distclean: doc-clean
 	rm -f $(doc_DISTCLEANFILES)
 
 doc-maintainer-clean: doc-distclean
 	rm -f $(doc_MAINTAINERCLEANFILES)
+	rm -rf $(DVIS) $(HTMLS) $(PDFS) $(PSS)
